### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by sprakash28**
@Behnke19 for review
Added Security Group operations with IHttpResponse to handle error scenarios to fix DE2149.

### CODE REVIEW

**Comment by servicemesh-autobuild** [comment]
:x: Build failed

http://ci.servicemesh.com:8080/job/com.servicemesh.agility.adapters.core.aws-prb/11/

File: ivy.xml [hunk 1]
```diff
@@ -1,54 +1,100 @@
...
+
+		<!-- common.ivy.lib -->
+		<dependency org="log4j" name="log4j" rev="1.2.17" transitive="false">
+			<artifact name="log4j" type="jar" ext="jar" />
+		</dependency>
+		<dependency org="org.slf4j" name="slf4j-log4j12" rev="1.7.5" transitive="false">
+			<artifact name="slf4j-log4j12" type="jar" ext="jar" />
+		</dependency>
+		<!-- common.dist -->
+		<dependency org="com.servicemesh" name="agility.api" rev="getLatestFromBintray(10,2,4)" transitive="false">
```
Comment (Behnke19) [inline ivy.xml]: There is an easier way to grab the latest. Make the following changes to the ivysettings.xml and ivy.xml files (see attached)
[ivyUpdates.txt](https://github.com/csc/com.servicemesh.agility.adapters.core.aws/files/493324/ivyUpdates.txt)

**Review by Behnke19 — CHANGES_REQUESTED** [review]

**Comment by sprakash28** [comment]
It seems this pr build is running on java 7

**Comment by Behnke19** [comment]
the pr build ran on our agility-java8 build machine. I updated it to be the agility-java8-prb machine to see if that makes a difference

**Comment by Behnke19** [comment]
retest

**Comment by servicemesh-autobuild** [comment]
:x: Build failed

http://ci.servicemesh.com:8080/job/com.servicemesh.agility.adapters.core.aws-prb/12/

**Comment by Behnke19** [comment]
retest

**Comment by servicemesh-autobuild** [comment]
:white_check_mark: Build was successful :tada:

http://ci.servicemesh.com:8080/job/com.servicemesh.agility.adapters.core.aws-prb/13/

**Comment by servicemesh-autobuild** [comment]
:x: Build failed

http://ci.servicemesh.com:8080/job/com.servicemesh.agility.adapters.core.aws-prb/14/

**Comment by servicemesh-autobuild** [comment]
:x: Build failed

http://ci.servicemesh.com:8080/job/com.servicemesh.agility.adapters.core.aws-prb/15/

**Comment by servicemesh-autobuild** [comment]
:white_check_mark: Build was successful :tada:

http://ci.servicemesh.com:8080/job/com.servicemesh.agility.adapters.core.aws-prb/16/

**Review by Behnke19 — APPROVED** [review]

**Pull request merged**